en/controllers/components/authentication.rst

```diff
@@ -563,7 +563,9 @@ database, the easiest way is to use a setter function in your User entity::
 
         protected function _setPassword($password)
         {
-            return (new DefaultPasswordHasher)->hash($password);
+            if (!empty($password)) {
```

Comment:
I think you want strlen() here. empty() will do incorrect things if the password is set to '0'.